File: 2019/day01.py
import numpy as np

# Part 1
m = np.loadtxt("day01_input.txt")
total_fuel = int(np.sum(np.floor(m/3)-2))
print("Part 1: total fuel is {:d}".format(total_fuel))

# Part 2
total_fuel = 0
for i in range(m.shape[0]):
    fm = 0
    f = int(np.floor(m[i]/3)-2)
    while (f > 0):
        fm = fm + f
        f = int(np.floor(f/3)-2)
    total_fuel = total_fuel + fm
print("Part 2: total fuel is {:d}".format(total_fuel))

File: 2019/day01.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main(int argc, char *argv[]) {

	FILE *fid = fopen("day01_input.txt","r");
	int fuel = 0;
	int f;
	while (!feof(fid)) {
		fscanf(fid, "%d", &f);
		fuel = fuel + floor(f/3)-2;
	}
	fclose(fid);
	printf("Part 1: fuel = %d\n", fuel);

	fid = fopen("day01_input.txt","r");
	int fm;
	fuel = 0;
	while (!feof(fid)) {
		fm = 0;
		fscanf(fid, "%d", &f);
		f = floor(f/3)-2;
		while (f > 0) {
			fm = fm + f;
			f = floor(f/3)-2;
		}
		fuel = fuel + fm;
	}
	fclose(fid);
	printf("Part 2: fuel = %d\n", fuel);

	return 0;
}